services/jwt-provider/provider.go

```diff
@@ -0,0 +1,213 @@
+package jwt_provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/forta-network/forta-core-go/clients/health"
+	"github.com/forta-network/forta-core-go/security"
+	"github.com/forta-network/forta-node/clients"
+	"github.com/forta-network/forta-node/config"
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+)
+
+// JWTProvider provides jwt tokens to bots, signed with node's private key..
+type JWTProvider struct {
+	botConfigs      []config.AgentConfig
+	botConfigsMutex sync.RWMutex
+
+	// to match request ip <-> bot id
+	dockerClient clients.DockerClient
+
+	cfg *JWTProviderConfig
+
+	lastErr health.ErrorTracker
+
+	srv *http.Server
+}
+
+type JWTProviderConfig struct {
+	Key    *keystore.Key
+	Config config.Config
+}
+
+func NewJWTProvider(
+	cfg config.Config,
+) (*JWTProvider, error) {
+	key, err := security.LoadKey(config.DefaultContainerKeyDirPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return initProvider(
+		&JWTProviderConfig{
+			Key:    key,
+			Config: cfg,
+		},
+	)
+}
+
+func initProvider(cfg *JWTProviderConfig) (*JWTProvider, error) {
+	globalClient, err := clients.NewDockerClient("")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create the global docker client: %v", err)
+	}
+
+	return &JWTProvider{dockerClient: globalClient, cfg: cfg}, nil
+}
+
+// Start spawns a jwt provider routine and returns.
+func (j *JWTProvider) Start() error {
+	return j.StartWithContext(context.Background())
+}
+
+func (j *JWTProvider) Stop() error {
+	return j.srv.Close()
+}
+
+// StartWithContext subscribe to bot updates and spawn a Bot JWT Provider http server.
+func (j *JWTProvider) StartWithContext(ctx context.Context) error {
+	if j.cfg.Config.JWTProvider.Addr == "" {
+		j.cfg.Config.JWTProvider.Addr = fmt.Sprintf(":%s", config.DefaultJWTProviderPort)
+	}
+
+	// setup routes
+	r := mux.NewRouter()
+	r.HandleFunc("/create", j.createJWTHandler).Methods(http.MethodPost)
+
+	j.srv = &http.Server{
+		Addr:    j.cfg.Config.JWTProvider.Addr,
```

Comment:
Let's use the hardcoded value if we remove `JWTProviderConfig` (mentioned in one of the other comments).